source/s3_default_meta_request.c

```diff
@@ -272,9 +267,8 @@ static void s_s3_default_prepare_request_on_read_done(void *user_data) {
         goto finish;
     }
 
-    /* TODO: support for unknown content length. move this check to read_body()? */
     if (request->request_body.len < request->request_body.capacity) {
-        error_code = aws_raise_error(AWS_ERROR_S3_INCORRECT_CONTENT_LENGTH_HEADER);
+        error_code = AWS_ERROR_S3_INCORRECT_CONTENT_LENGTH_HEADER;
```

Comment:
Do me a biggie and shorten the name of this new error
```suggestion
        error_code = AWS_ERROR_S3_INCORRECT_CONTENT_LENGTH;
```